plugin.video.greekvoice/resources/lib/runner.py

```diff
@@ -0,0 +1,187 @@
+#  Greek Voice Add-on
+#  Author: threshold84
+#  SPDX-License-Identifier: GPL-3.0
+#  See LICENSES/GPL-3.0 for more information.
+
+import os
+import sys
+
+import xbmc
+import xbmcvfs
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+from .url_dispatcher import urldispatcher
+from .scrapetube import wrapper
+from urllib.parse import quote_plus
+
+__addon__ = xbmcaddon.Addon()
+__lang__ = __addon__.getLocalizedString
+__handle__ = int(sys.argv[1])
+
+
+ADDON_ID = __addon__.getAddonInfo('id')
+HOME = xbmcvfs.translatePath(__addon__.getAddonInfo('path'))
+ART_FOLDER = os.path.join(HOME, "resources", "art")
+ICON = os.path.join(ART_FOLDER, 'icon.png')
+FANART = os.path.join(ART_FOLDER, 'fanart.jpg')
+
+GREEK_VOICE_URL_1 = 'http://65.108.62.57:8080/live/greektv1.m3u8'
+GREEK_VOICE_URL_2 = 'http://wpso.com:1936/hls/wzra.m3u8'
+TILEMOUSIKI_URL_1 = 'http://65.108.62.57:8080/live/greekmusic.m3u8'
+TILEMOUSIKI_URL_2 = 'http://wpso.com:1936/hls/music.m3u8'
+CARTOONS_URL_1 = 'http://65.108.62.57:8080/live/kidshd.m3u8'
+CARTOONS_URL_2 = 'http://wpso.com:1936/hls/kidshd.m3u8'
+WPSO_GREEK_VOICE_RADIO_URL = 'http://wpso.com:8000/'
+WXYB_RADIO_GR_IT_ES = 'http://wpso.com:7071/'
+XAMOS_URL = 'http://xamosam.com:9050'
+YT_CHANNEL_URL = 'https://www.youtube.com/@GreekVoiceTV/'
+
+
+@urldispatcher.register('root')
+def root():
+
+    menu = [
+        {
+            'title': __lang__(30001),
+            'icon': os.path.join(ART_FOLDER, 'video_streams.jpg'),
+            'fanart': os.path.join(ART_FOLDER, 'video_streams.jpg'),
+            'kind': 'video'
+        }
+        ,
+        {
+            'title': __lang__(30002),
+            'icon': os.path.join(ART_FOLDER, 'radio_streams.jpg'),
+            'fanart': os.path.join(ART_FOLDER, 'radio_fanart.jpg'),
+            'kind': 'audio'
+        }
+        ,
+        {
+            'title': __lang__(30003),
+            'icon': os.path.join(ART_FOLDER, 'youtube.jpg'),
+            'fanart': FANART,
+            'kind': 'youtube'
+        }
+    ]
+
+    for m in menu:
+        list_item = xbmcgui.ListItem(m['title'])
```

Comment:
might consider to add offscreen=True as argument in the initializer (avoid gui locks)

Reply:
Done